src/Http/Middleware/ReferrerPolicyHeader.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Http\Middleware;
+
+use Flarum\Foundation\Config;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface as Middleware;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class ReferrerPolicyHeader implements Middleware
+{
+    protected $policy = '';
+
+    public function __construct(Config $config)
+    {
+        $this->policy = $config['headers']['referrer-policy'] ?? 'same-origin';
```

Comment:
Do we use `config['headers']` for anything else? If we choose to use it here, we should move the promotional header key to use that as well.

Reply:
We currently don't but I think we might want too?

Reply:
I'm in favor, but let's add a PR to change the promotional header to use that too